src/plugins/visualizations/public/vis_types/base_vis_type.ts

```diff
@@ -17,118 +17,113 @@
-  editorConfig?: Record<string, any>;
-  hidden?: boolean;
-  requestHandler?: string | unknown;
-  responseHandler?: string | unknown;
-  hierarchicalData?: boolean | unknown;
-  setup?: unknown;
-  useCustomNoDataScreen?: boolean;
-  inspectorAdapters?: Adapters | (() => Adapters);
-  isDeprecated?: boolean;
-  getDeprecationMessage?: (vis: Vis) => ReactElement<{}>;
+interface CommonBaseVisTypeOptions<TVisParams>
+  extends Pick<
+      VisType<TVisParams>,
+      | 'description'
+      | 'editor'
+      | 'getDeprecationMessage'
+      | 'getSupportedTriggers'
+      | 'hierarchicalData'
+      | 'icon'
+      | 'image'
+      | 'inspectorAdapters'
+      | 'name'
+      | 'requestHandler'
+      | 'responseHandler'
+      | 'setup'
+      | 'title'
+    >,
+    Pick<
+      Partial<VisType<TVisParams>>,
+      'editorConfig' | 'hidden' | 'stage' | 'useCustomNoDataScreen' | 'visConfig'
+    > {
+  options?: Partial<VisType<TVisParams>['options']>;
 }
 
-interface ExpressionBaseVisTypeOptions<TVisParams> extends CommonBaseVisTypeOptions {
-  toExpressionAst: VisToExpressionAst<TVisParams>;
+interface ExpressionBaseVisTypeOptions<TVisParams> extends CommonBaseVisTypeOptions<TVisParams> {
+  toExpressionAst: VisType<TVisParams>['toExpressionAst'];
   visualization?: undefined;
 }
 
-interface VisualizationBaseVisTypeOptions extends CommonBaseVisTypeOptions {
+interface VisualizationBaseVisTypeOptions<TVisParams> extends CommonBaseVisTypeOptions<TVisParams> {
   toExpressionAst?: undefined;
-  visualization: VisualizationControllerConstructor | undefined;
+  visualization: VisType<TVisParams>['visualization'];
 }
 
 export type BaseVisTypeOptions<TVisParams = VisParams> =
   | ExpressionBaseVisTypeOptions<TVisParams>
-  | VisualizationBaseVisTypeOptions;
+  | VisualizationBaseVisTypeOptions<TVisParams>;
 
-export class BaseVisType<TVisParams = VisParams> {
-  name: string;
-  title: string;
-  description: string;
-  getSupportedTriggers?: () => Array<keyof TriggerContextMapping>;
-  icon?: string;
-  image?: string;
-  stage: 'experimental' | 'beta' | 'production';
-  isExperimental: boolean;
-  options: Record<string, any>;
-  visualization: VisualizationControllerConstructor | undefined;
-  visConfig: Record<string, any>;
-  editor: any;
-  editorConfig: Record<string, any>;
-  hidden: boolean;
-  requiresSearch: boolean;
-  requestHandler: string | unknown;
-  responseHandler: string | unknown;
-  hierarchicalData: boolean | unknown;
-  setup?: unknown;
-  useCustomNoDataScreen: boolean;
-  inspectorAdapters?: Adapters | (() => Adapters);
-  toExpressionAst?: VisToExpressionAst<TVisParams>;
-  getDeprecationMessage?: (vis: Vis) => ReactElement<{}>;
+const defaultOptions: VisTypeOptions = {
+  showTimePicker: true,
+  showQueryBar: true,
+  showFilterBar: true,
+  showIndexSelection: true,
+  hierarchicalData: false, // we should get rid of this i guess ?
+};
+
+export class BaseVisType<TVisParams = VisParams> implements VisType<TVisParams> {
```

Comment:
:information_source: Those now properly `implement` the interface and thus you also don't need to type the properties here again, since it can pull it automatically from the interface.

Reply:
Nice 🙂